packages/block-library/src/column/save.js

```diff
@@ -16,8 +16,10 @@ export default function save( { attributes } ) {
 	} );
 
 	let style;
+
 	if ( width ) {
-		style = { flexBasis: width };
+		// Numbers are handled for backward compatibility as they can be still provided with templates.
+		style = { flexBasis: Number.isFinite( width ) ? width + '%' : width };
```

Comment:
I will explore it further, but seeing this PR again, I started thinking about whether new changes applied to `edit` are enough. Anyway, I can send a follow-up PR when I confirm it isn't necessary anymore. For now, better safe than sorry 😄

Reply:
I tested it with the following diff:

```diff
diff --git a/packages/block-library/src/column/save.js b/packages/block-library/src/column/save.js
index c43edc8464..ccf1b64387 100644
--- a/packages/block-library/src/column/save.js
+++ b/packages/block-library/src/column/save.js
@@ -18,8 +18,7 @@ export default function save( { attributes } ) {
 	let style;
 
 	if ( width ) {
-		// Numbers are handled for backward compatibility as they can be still provided with templates.
-		style = { flexBasis: Number.isFinite( width ) ? width + '%' : width };
+		style = { flexBasis: width };
 	}
 
 	return (
diff --git a/packages/block-library/src/columns/variations.js b/packages/block-library/src/columns/variations.js
index 9f20699dee..6026a5f51e 100644
--- a/packages/block-library/src/columns/variations.js
+++ b/packages/block-library/src/columns/variations.js
@@ -74,8 +74,8 @@ const variations = [
 			</SVG>
 		),
 		innerBlocks: [
-			[ 'core/column', { width: '33.33%' } ],
-			[ 'core/column', { width: '66.66%' } ],
+			[ 'core/column', { width: 33.33 } ],
+			[ 'core/column', { width: 66.66 } ],
 		],
 		scope: [ 'block' ],
 	},
```

and it seems to work correctly. Do you think we should update codebase or do I miss some other case?

Reply:
Tested this by adding a 30/70 columns block, and both columns are rendering with the same size. But that happens with or without your change to `save.js`, so looking for the issue now 🤔

Reply:
Scratch that, I was passing the template widths in as unitless strings 😅 . That's not something we'd ever expect to happen, is it?

I think it's fine to make this change, anyway!

Reply:
Yes, it works properly only with numbers. I had the same issue with strings. We can always check the issue filed and previous setup before it was converted to percentages. If it worked with strings in the past, we might need another fix 😃